1517: Build search box component

Caleb-Ellis opened the pull request:
## Done

- Built search box component as per [specs](https://github.com/ubuntudesign/vanilla-design/tree/master/Search)
- Includes Default and Navigation examples
- Added docs page for the search box component (and included it in the side nav this time)

## QA

- Pull code
- Run `./run`
- Open http://0.0.0.0:8101/vanilla-framework/examples/patterns/search-box/default and http://0.0.0.0:8101/vanilla-framework/examples/patterns/search-box/navigation
- Check that the component matches the [specs](https://github.com/ubuntudesign/vanilla-design/tree/master/Search)
- Check that everything looks right in all browsers that we support ([link](https://github.com/canonical-webteam/practices/blob/master/coding/browser-support.md))
- `cd` into the docs folder and run `./run`
- Open http://0.0.0.0:8104/
- Check that you can find 'Search box' in the side nav under 'Interactive elements'
- Check that it navigates to the appropriate docs page

## Details

Fixes #1517

webteam-app commented:
Starting demo at: http://vanilla-framework-pr-1521.run.demo.haus/vanilla-framework/

kwm14 commented:
Looking good @Caleb-Ellis 👍

barrymcgee commented on docs/en/patterns/search-box.md:
```diff
@@ -0,0 +1,13 @@
+---
+title: Search box
+table_of_contents: true
```
Does this need a ToC if it only has one heading?

Caleb-Ellis commented:
@barrymcgee Great points. I've added some search-box scss to `.p-navigation` so it should drop in nicely now, and added a new example to show it in action.

barrymcgee commented:
Thanks @Caleb-Ellis, looks good to me.

spencerbygraves commented:
@Caleb-Ellis @kwm14 Looking good, just a couple of things I've noticed, apologies if they are expected.

There is no space above the search box, looks like we will need a version of the navigation with more vertical space, or a search field with less vertical space.

![image](https://user-images.githubusercontent.com/9105999/35091639-00c8cca4-fc35-11e7-877f-63a7b009e28a.png)

Also, on small screen there is no space, and the X icon is missing as in the specs, or will that be in a separate PR?

![image](https://user-images.githubusercontent.com/9105999/35091738-473d9534-fc35-11e7-9d9d-c029f6fc5aec.png)


Thanks

webteam-app commented:
Starting demo at: http://vanilla-framework-pr-1521.run.demo.haus/vanilla-framework/

kwm14 commented:
@Caleb-Ellis agree with @spencerbygraves we need some padding on top of the search box so it doesn't sit on top of the navigation bar.

- [ ] Add `0.5rem` padding top to match the padding at the bottom

Also on small screen are we able to keep it in the top navigation? If not can we move it to the top of the nav menu rather than at the bottom...

![screen shot 2018-01-18 at 10 30 14](https://user-images.githubusercontent.com/17748020/35093603-b995ba1c-fc3a-11e7-836c-8131e94fb38a.png)

Caleb-Ellis commented:
@spencerbygraves agreed about the top padding. This should be fixed now. For the X, I've made it so it only shows up if there's text in the search form.

@kwm14 search box is now vertically centred. As for the position on small screens, it's a bit tricky to have it at the very top banner because it sits in a different DOM node (banner instead of links). It's possible to have it in the top banner if we have two search boxes, one hidden on small screens and the other hidden on large screens but I think it might over complicate things. For now I've just made it sit at the top of the links.

webteam-app commented:
Starting demo at: http://vanilla-framework-pr-1521.run.demo.haus/vanilla-framework/

kwm14 commented:
@Caleb-Ellis thanks for updating. Looks good 👍 

Shame we can't have the search inside the top nav on smaller screens but like you say:
- it's possible to have it in the top banner if we have two search boxes
- one hidden on small screens and the other hidden on large screens

I'll create an item in the backlog for an enhancement for the search box which I can design and spec to be built.
